### PR TITLE
Eperez phone ms

### PR DESCRIPTION
**Pull request by enriquepablo**
Phone proofing elements

### CODE REVIEW

File: src/eduid_userdb/proofing/proofingdb.py [hunk 1]
```diff
@@ -250,6 +286,55 @@ def remove_state(self, state):
...
+
+
+class PhoneProofingStateDB(EmailProofingStateDB):
+
+    ProofingStateClass = PhoneProofingState
+
+    def __init__(self, db_uri, db_name='eduid_phones'):
+        ProofingStateDB.__init__(self, db_uri, db_name)
+
+    def get_state_by_mail_and_code(self, mail, code, raise_on_missing=True):
```
Comment (johanlundberg) [inline src/eduid_userdb/proofing/proofingdb.py]: This method seems out of place?

File: src/eduid_userdb/proofing/proofingdb.py [hunk 2]
```diff
@@ -250,6 +286,55 @@ def remove_state(self, state):
...
+        """
+        :param state: ProofingStateClass object
+
+        :type state: ProofingStateClass
+        """
+        self.remove_document({'eduPersonPrincipalName': state.eppn,
+                              'verification.number': state.verification.number})
+
+
+class PhoneProofingStateDB(EmailProofingStateDB):
```
Comment (johanlundberg) [inline src/eduid_userdb/proofing/proofingdb.py]: Do PhoneProofingStateDB need to inherit from EmailProofingStateDB? Isn't ProofingStateDB enough?

File: src/eduid_userdb/proofing/proofingdb.py [hunk 3]
```diff
@@ -250,6 +286,55 @@ def remove_state(self, state):
         self.remove_document({'eduPersonPrincipalName': state.eppn,
                         'verification.email': state.verification.email})
 
+    def remove_state_phone(self, state):
```
Comment (johanlundberg) [inline src/eduid_userdb/proofing/proofingdb.py]: I think this should be moved to PhoneProofingStateDB and just called remove_state.

File: src/eduid_userdb/proofing/proofingdb.py [hunk 4]
```diff
@@ -241,6 +242,41 @@ def get_state_by_eppn_and_email(self, eppn, email, raise_on_missing=True):
             # and [].count() raises a TypeError
             return None
 
+    def get_state_by_eppn_and_mobile(self, eppn, number, raise_on_missing=True):
```
Comment (johanlundberg) [inline src/eduid_userdb/proofing/proofingdb.py]: I think this should be moved to PhoneProofingStateDB.

File: src/eduid_userdb/phone.py [hunk 5]
```diff
@@ -48,16 +48,30 @@ class PhoneNumber(PrimaryElement):
...
-    def __init__(self, data, raise_on_unknown = True):
+    def __init__(self, number=None, application=None, verified=False, created_ts=None,
+                 primary=False, verification_code=None, data=None, raise_on_unknown = True):
         data_in = data
         data = copy.copy(data_in)  # to not modify callers data
 
+        if data is None:
+            if created_ts is None:
+                created_ts = True
+            # TODO: Created_by = application
```
Comment (johanlundberg) [inline src/eduid_userdb/phone.py]: Is the todo done or am I misunderstanding it?

File: src/eduid_userdb/proofing/proofingdb.py (hunk 2 above)
Comment (enriquepablo) [inline src/eduid_userdb/proofing/proofingdb.py]: Yes, this inheritance is quite crazy, I'll fix it.

File: src/eduid_userdb/phone.py (hunk 5 above)
Comment (enriquepablo) [inline src/eduid_userdb/phone.py]: Yes it's done, removing the comment.

**Pull request merged**